### PR TITLE
PMM-5917 check RELOAD privilege before log rotation

### PR DESCRIPTION
**Pull request by palash25**
https://jira.percona.com/browse/PMM-5917

### CODE REVIEW

File: agents/mysql/slowlog/slowlog.go [hunk 1]
```diff
@@ -163,7 +164,25 @@ func (s *SlowLog) Run(ctx context.Context) {
 
 // recheck returns new slowlog information, and rotates slowlog file if needed.
 func (s *SlowLog) recheck(ctx context.Context) (newInfo *slowLogInfo) {
-	var err error
+	db, err := sql.Open("mysql", s.params.DSN)
+	if err != nil {
+		s.l.Errorf("Cannot open database connection: %s", err)
+		return
+	}
+	defer db.Close() //nolint:errcheck
```
Comment (oter) [inline agents/mysql/slowlog/slowlog.go:172]: can we log the error?

File: agents/mysql/slowlog/slowlog.go [hunk 2]
```diff
@@ -163,7 +164,25 @@ func (s *SlowLog) Run(ctx context.Context) {
 
 // recheck returns new slowlog information, and rotates slowlog file if needed.
 func (s *SlowLog) recheck(ctx context.Context) (newInfo *slowLogInfo) {
-	var err error
+	db, err := sql.Open("mysql", s.params.DSN)
+	if err != nil {
+		s.l.Errorf("Cannot open database connection: %s", err)
```
Comment (oter) [inline agents/mysql/slowlog/slowlog.go:169]: maybe we can return such errors and log them on the caller's side?

**Review by percona-csalguero — APPROVED** [review]

File: agents/mysql/slowlog/slowlog.go (hunk 2 above)
Comment (palash25) [reply agents/mysql/slowlog/slowlog.go:169]: disagree,  if we return the error and still just log it then its not much different from the current code. we will also have to change the function signature and the return statements everywhere in `recheck` better to prefer a smaller code change over a slightly bigger one if both do the same thing

**Review by jprukner — APPROVED** [review]
Were docs updated to include this privileges?

**Review by oter — APPROVED** [review]

**Comment by palash25** [comment]
> Were docs updated to include the privilege?

the docs were already there, it was this check that was missing

**Pull request merged**